src/function.ts

```diff
@@ -0,0 +1,166 @@
+import { Config, Connection } from "./connection";
+
+export interface DeployQueryWorkerParameters {
+  type: string;
+  name: string;
+  queryWorkerName: string;
+  environment: string;
+}
+
+export interface DeployStreamAdhocQueryParameters {
+  type: string;
+  name: string;
+  streamWorkerName: string;
+  environment: string;
+}
+
+export interface DeployStreamPublisherParameters {
+  type: string;
+  name: string;
+  streamWorkerName: string;
+  streamName: string;
+  environment: string;
+}
+
+export interface MetadataParameters {
+  type: string;
+  accessToken: string;
+  baseUri: string;
+  clientSecret: string;
+  clientToken: string;
+  resourceTierId: string;
+  groupId: string;
+  hostName: string;
+}
+
+export class Function {
+  protected _connection: Connection;
+
+  constructor(config?: Config) {
+    this._connection = new Connection(config);
+  }
+  listFunctionWorkers(type: string = "all") {
+    return this._connection.request(
+      {
+        method: "GET",
+        path: "/_api/function",
+        qs: type,
+      },
+      res => res.body
+    );
+  }
+
+  deployQueryWorkerToEdgeWorker(parameters: DeployQueryWorkerParameters) {
+    return this._connection.request(
+      {
+        method: "POST",
+        path: "/_api/function/generate",
+        qs: parameters,
+      },
+      res => res.body
+    );
+  }
+
+  deployStreamAdhocQueryToEdgeWorker(
+    parameters: DeployStreamAdhocQueryParameters
+  ) {
+    return this._connection.request(
+      {
+        method: "POST",
+        path: "/_api/function/generate/query",
+        qs: parameters,
+      },
+      res => res.body
+    );
+  }
+
+  deployStreamPublisherToEdgeWorker(
+    parameters: DeployStreamPublisherParameters
+  ) {
+    return this._connection.request(
+      {
+        method: "POST",
+        path: "/_api/function/generate/publisher",
+        qs: parameters,
+      },
+      res => res.body
+    );
+  }
+
+  getFunctionWorkerInfo(functionName: string) {
+    return this._connection.request(
+      {
+        method: "GET",
+        path: `/_api/function/${functionName}`,
+      },
+      res => res.body
+    );
+  }
+
+  removeFunctionWorker(functionName: string) {
+    return this._connection.request(
+      {
+        method: "DELETE",
+        path: `/_api/function/${functionName}`,
+      },
+      res => res.body
+    );
+  }
+
+  invokeFunctionWorker(functionName: string, parameters?: any) {
```

Comment:
if we have know set of parameters then it's better to create a type.

Reply:
This is tricky @VikasPatel, actually we don't know the set of parameters. Those are bind variables for queries, users can name them and bind values as they saw fit. That is why I am using `any`. It is not suggested, but I can't see any other solution for unknown parameters.